parachain/pallets/incentivized-channel/src/outbound/benchmarking.rs

```diff
@@ -0,0 +1,65 @@
+//! IncentivizedOutboundChannel pallet benchmarking
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::*;
+
+use frame_benchmarking::{benchmarks, impl_benchmark_test_suite};
+use frame_support::traits::OnInitialize;
+
+#[allow(unused_imports)]
+use crate::outbound::Module as IncentivizedOutboundChannel;
+
+benchmarks! {
+	// Benchmark `on_initialize` under worst case conditions, i.e. messages
+	// in queue are committed.
+	on_initialize {
+		let m in 1 .. T::MaxMessagesPerCommit::get() as u32;
+
+		for _ in 0 .. m {
+			MessageQueue::append(Message {
+				target: H160::zero(),
+				nonce: 0u64,
+				payload: vec![1, 2, 3],
```

Comment:
We should benchmark using a bigger payload, as the payload size does affect the weight. As part of the commitment generation process, the payload is hashed and that is fairly expensive.

Looking over our current set of apps, they all generate payloads roughly less than 128 bytes. For example:  https://github.com/Snowfork/polkadot-ethereum/blob/1978dfb5b49b68618ee87172500c0e567697cce7/parachain/pallets/dot-app/src/lib.rs#L127

So may be try a payload like `vec![1u8; 128]`.

Reply:
Good point. I'm going to try make the payload size a variable and see how it goes